File: backend/PersonalLibrary.Api/Models/Book.cs
using System.ComponentModel.DataAnnotations;

namespace PersonalLibrary.Api.Models;

public class Book
{
    public int Id { get; set; }

    [Required, MaxLength(200)]
    public string Title { get; set; } = string.Empty;

    [MaxLength(200)]
    public string? Author { get; set; }

    [Required]
    public string Description { get; set; } = string.Empty;

    [Required]
    public string CoverImageUrl { get; set; } = string.Empty;

    public int? Year { get; set; }


}

File: frontend/src/types/book.ts
export type ReadingStatus = "to-read" | "reading" | "finished";

export interface Book {
  id: string;
  title: string;
  author: string;
  cover: string;
  coverImageUrl?: string;

  rating: number;
  category: string;
  description: string;
  pages: number;
  publishedYear: number;
  year?: number;
  duration?: string;
  isAudiobook: boolean;

  // ✅ NEW: where to read as PDF
  pdfUrl?: string;
  previewUrl?: string;

}

export interface UserBook extends Book {
  status: ReadingStatus;
  progress: number;
  currentPage?: number;
  pagesRead?: number;
  totalPages?: number;
  percent?: number;
  lastUpdated?: string;
  userRating?: number;
  dateAdded: string;
}

export interface User {
  id: string;
  name: string;
  avatar: string;
  booksCount: number;
  recentBooks: Book[];
}
